fleetspeak/src/e2etesting/balancer/balancer.go

```diff
@@ -23,6 +23,18 @@ func copy(wc io.WriteCloser, r io.Reader) {
 	io.Copy(wc, r)
 }
 
+func renderProxyV1Header(srcAddr, dstAddr string) ([]byte, error) {
+	srcColPos := strings.Index(srcAddr, ":")
+	if srcColPos == -1 {
+		return nil, fmt.Errorf("Failed to parse source (client) address")
+	}
+	dstColPos := strings.Index(dstAddr, ":")
+	if dstColPos == -1 {
+		return nil, fmt.Errorf("Failed to parse destination (server) address")
+	}
+	return []byte(fmt.Sprintf("PROXY TCP4 %v %v %v %v\r\n", srcAddr[:srcColPos], dstAddr[:dstColPos], srcAddr[srcColPos+1:len(srcAddr)], dstAddr[dstColPos+1:len(dstAddr)])), nil
```

Comment:
I'd probably use the go-proxyproto function - adds IP6 support, and is otherwise more likely to get corner cases right.

https://github.com/pires/go-proxyproto/blob/master/header.go#L99

Reply:
+1